core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -87,10 +94,13 @@ final class ShuffleBlockFetcherIterator(
     maxBlocksInFlightPerAddress: Int,
     val maxReqSizeShuffleToMem: Long,
     maxAttemptsOnNettyOOM: Int,
+    slowFetchThresholdMs: Long,
+    slowFetchThresholdBps: Long,
     detectCorrupt: Boolean,
     detectCorruptUseExtraMemory: Boolean,
     shuffleMetrics: ShuffleReadMetricsReporter,
-    doBatchFetch: Boolean)
+    doBatchFetch: Boolean,
+    clock: Clock = new SystemClock())
```

Comment:
If we're adding a clock, we should use it for all timing information within the class. How big would the diff be to move all uses of `System`-based timing to the `clock`?

Reply:
There were only a couple of usages of `System`, so I replaced them.